show/reboot_cause.py

```diff
@@ -8,15 +8,19 @@
 import utilities_common.cli as clicommon
 
 
-PREVIOUS_REBOOT_CAUSE_FILE = "/host/reboot-cause/previous-reboot-cause.json"
-USER_ISSUED_REBOOT_CAUSE_REGEX ="User issued \'{}\' command [User: {}, Time: {}]"
+PREVIOUS_REBOOT_CAUSE_FILE_PATH = "/host/reboot-cause/previous-reboot-cause.json"
 
 def read_reboot_cause_file():
-    result = ""
-    if os.path.exists(PREVIOUS_REBOOT_CAUSE_FILE):
-        with open(PREVIOUS_REBOOT_CAUSE_FILE) as f:
-            result = json.load(f)
-    return result
+    reboot_cause_dict = {}
+
+    if os.path.exists(PREVIOUS_REBOOT_CAUSE_FILE_PATH):
+        with open(PREVIOUS_REBOOT_CAUSE_FILE_PATH) as prev_reboot_cause_file:
+            try:
+                reboot_cause_dict = json.load(prev_reboot_cause_file)
+            except json.JSONDecodeError as err:
+                click.echo("Failed to load JSON file '{}'!".format(PREVIOUS_REBOOT_CAUSE_FILE_PATH))
```

Comment:
@jleveque If we can not read the JSON file successfully, then CLI `show reboot-cause` will output:

    Failed to load JSON file /host/reboot-cause/previous-reboot-cause.json!
    Unknow

I am wondering whether it is appropriate?

Reply:
Under what circumstances are you seeing this? `/host/reboot-cause/previous-reboot-cause.json` should be written by the determine-reboot-cause service which is one of the first services run after boot.

Reply:
If the service `determine-reboot-cause` can not write the reboot reason into the file `/host/reboot-cause/previous-reboot-cause.json` successfully, then this JSON file will be an empty file. If we run this CLI `show reboot-cause`, this exception will be raised.

Reply:
Under what circumstances will `determine-reboot-cause` fail to write the file?

Reply:
One possible reason I think is the process `determine-reboot-cause` opened the file, but it crashed before it can write previous reboot reason into the file.

Reply:
In that case, we would need to fix a bug in `determine-reboot-cause`. I think `Unknown` is the proper thing to display in this case, and outputting an error would be helpful to understand why. Maybe we should simply send the error message to stderr:

```python
click.echo("Failed to load JSON file '{}'!".format(PREVIOUS_REBOOT_CAUSE_FILE_PATH), err=True)
```